core/src/main/scala/kafka/log/Log.scala

```diff
@@ -1784,8 +1784,18 @@ class Log(@volatile private var _dir: File,
   private def deleteRetentionMsBreachedSegments(): Int = {
     if (config.retentionMs < 0) return 0
     val startMs = time.milliseconds
-    deleteOldSegments((segment, _) => startMs - segment.largestTimestamp > config.retentionMs,
-      reason = s"retention time ${config.retentionMs}ms breach")
+
+    def shouldDelete(segment: LogSegment, nextSegmentOpt: Option[LogSegment]) = {
+      if (startMs - segment.largestTimestamp > config.retentionMs) {
+        info(s"Segment with base offset ${segment.baseOffset} will be deleted due to" +
+          s" retentionMs breach. Largest timestamp of segment is ${segment.largestTimestamp}")
+        true
+      } else {
+        false
+      }
+    }
+
+    deleteOldSegments(shouldDelete, reason = s"retention time ${config.retentionMs}ms breach")
```

Comment:
With the logging we have above, do you think we still need the message in `deleteOldSegments`? Perhaps we could make it more concise at least. Maybe just mention the number of segments to be deleted for example.

Reply:
Yeah I agree, I think it would be redundant to keep the `reason` in `deleteOldSegments`. I'll remove that and as you suggested just mentioned the number of segments getting deleted.